operator/builtin/output/otlp/converter.go

```diff
@@ -0,0 +1,244 @@
+	"github.com/observiq/stanza/version"
+	"go.opentelemetry.io/collector/consumer/pdata"
+)
+
+// Convert converts a slice of entries to pdata.Logs format
+func Convert(entries []*entry.Entry) pdata.Logs {
+
+	out := pdata.NewLogs()
+	logs := out.ResourceLogs()
+
+	entriesByResource := groupByResource(entries)
+
+	logs.Resize(len(entriesByResource))
+
+	for i, resourceEntries := range entriesByResource {
+		rls := logs.At(i)
+		resource := rls.Resource()
+		resource.InitEmpty()
+
+		resourceAtts := resource.Attributes()
+		for k, v := range resourceEntries[0].Resource {
+			resourceAtts.InsertString(k, v)
+		}
+
+		rls.InstrumentationLibraryLogs().Resize(1)
+		ills := rls.InstrumentationLibraryLogs().At(0)
+		ills.InitEmpty()
+
+		il := ills.InstrumentationLibrary()
+		il.InitEmpty()
+		il.SetName("stanza")
+		il.SetVersion(version.GetVersion())
+
+		for _, entry := range resourceEntries {
+			lr := pdata.NewLogRecord()
+			lr.InitEmpty()
+			lr.SetTimestamp(pdata.TimestampUnixNano(entry.Timestamp.UnixNano()))
+
+			sevText, sevNum := convertSeverity(entry.Severity)
+			lr.SetSeverityText(sevText)
+			lr.SetSeverityNumber(sevNum)
+
+			if len(entry.Labels) > 0 {
+				attributes := lr.Attributes()
+				for k, v := range entry.Labels {
+					attributes.InsertString(k, v)
+				}
+			}
+
+			lr.Body().InitEmpty()
+			insertToAttributeVal(entry.Record, lr.Body())
+
+			ills.Logs().Append(lr)
+		}
+	}
+
+	return out
+}
+
+func groupByResource(entries []*entry.Entry) [][]*entry.Entry {
+	resourceMap := make(map[string][]*entry.Entry)
+
+	for _, ent := range entries {
+		resourceBytes, err := json.Marshal(ent.Resource)
+		if err != nil {
+			continue // not expected to ever happen
+		}
+		resourceHash := string(resourceBytes)
+
+		if resourceEntries, ok := resourceMap[resourceHash]; ok {
+			resourceEntries = append(resourceEntries, ent)
+		} else {
+			resourceMap[resourceHash] = []*entry.Entry{ent}
+		}
+	}
+
+	entriesByResource := make([][]*entry.Entry, 0, len(resourceMap))
+	for _, v := range resourceMap {
+		entriesByResource = append(entriesByResource, v)
+	}
+	return entriesByResource
+}
+
+func insertToAttributeVal(value interface{}, dest pdata.AttributeValue) {
+	switch t := value.(type) {
+	case bool:
+		dest.SetBoolVal(t)
+	case string:
+		dest.SetStringVal(t)
+	case []byte:
+		dest.SetStringVal(string(t))
+	case int64:
+		dest.SetIntVal(t)
+	case int32:
+		dest.SetIntVal(int64(t))
+	case int16:
+		dest.SetIntVal(int64(t))
+	case int8:
+		dest.SetIntVal(int64(t))
+	case int:
+		dest.SetIntVal(int64(t))
+	case uint64:
+		dest.SetIntVal(int64(t))
+	case uint32:
+		dest.SetIntVal(int64(t))
+	case uint16:
+		dest.SetIntVal(int64(t))
+	case uint8:
+		dest.SetIntVal(int64(t))
+	case uint:
+		dest.SetIntVal(int64(t))
+	case float64:
+		dest.SetDoubleVal(t)
+	case float32:
+		dest.SetDoubleVal(float64(t))
+	case map[string]interface{}:
+		dest.SetMapVal(toAttributeMap(t))
+	case []interface{}:
+		dest.SetArrayVal(toAttributeArray(t))
+	default:
```

Comment:
Any concerns about `map[string]string` et al?

Reply:
also `[]string` and friends

Reply:
There are definitely some that we'll miss, but I think those two at least are common enough that they should be added. 👍